File: frontend/src/types/task.ts
export interface Task {
  id: number;
  user_id: number;
  status: "queued" | "processing" | "ready_for_review" | "completed" | "failed";
  video_path: string;
  output_path: string | null;
  subtitle_path: string | null;
  progress: number;
  stage: string | null;
  error_message: string | null;
  created_at: string;
  completed_at: string | null;
}

export interface TaskCreate {
  video_path: string;
  asr_model?: string;
  config?: Record<string, unknown>;
}

export interface QueueStatus {
  queue_length: number;
  current_task: Task | null;
}

export interface SubtitleSegment {
  text: string;
  start_time: number;
  end_time: number;
  translated_text: string | null;
}

File: backend/app/schemas/task.py
"""Task request/response schemas."""

import datetime
from typing import Optional

from pydantic import BaseModel


class TaskCreate(BaseModel):
    video_path: str
    asr_model: str = "whisper_cpp"
    whisper_model: str = "base"
    config: dict = {}


class TaskResponse(BaseModel):
    id: int
    user_id: int
    status: str
    video_path: str
    output_path: Optional[str] = None
    subtitle_path: Optional[str] = None
    progress: int = 0
    stage: Optional[str] = None
    error_message: Optional[str] = None
    created_at: datetime.datetime
    completed_at: Optional[datetime.datetime] = None


class QueueStatus(BaseModel):
    queue_length: int
    current_task: Optional[TaskResponse] = None
